core/ir/disassemble_shared.c

```diff
@@ -1164,6 +1224,48 @@ sign_extend_immed(instr_t *instr, int srcnum, opnd_t *src)
     }
 }
 
+#define MAX_LENGTH_ALL_CATEGORIES_AS_STRING 57
+
+/* Returns a space-separated string representation of the list of categories an
+ * instruction belongs to.
+ */
+void
+get_category_names(uint category, char category_names[], size_t category_names_size)
+{
+    memset(category_names, '\0', category_names_size);
+    if (category == DR_INSTR_CATEGORY_UNCATEGORIZED) {
+        const char *category_name = instr_get_category_name(category);
+        size_t char_to_copy = strlen(category_name);
+        strncpy(category_names, category_name, char_to_copy);
+        return;
+    }
+
+    /* We consider 0x80000000 enough to be future proof when adding new categories.
+     */
+    const uint max_mask = 0x80000000;
+    uint start_index = 0;
+    for (uint mask = 0x1; mask <= max_mask; mask <<= 1) {
+        if (TESTANY(mask, category)) {
+            const char *category_name = instr_get_category_name(mask);
+            size_t char_to_copy = strlen(category_name);
+            strncpy(category_names + start_index, category_name, char_to_copy);
```

Comment:
This should be able to use print_to_buffer() which handles all the updating of appending to a buffer.

Reply:
Using `print_to_buffer()`. Done.